src/couch_replicator_manager.erl

```diff
@@ -124,13 +127,30 @@ replication_error(#rep{id = {BaseId, _} = RepId}, Error) ->
     nil ->
         ok;
     #rep_state{rep = #rep{db_name = DbName, doc_id = DocId}} ->
+        add_error_jitter(RepId, DbName, DocId),
         update_rep_doc(DbName, DocId, [
             {<<"_replication_state">>, <<"error">>},
             {<<"_replication_state_reason">>, to_binary(error_reason(Error))},
             {<<"_replication_id">>, ?l2b(BaseId)}]),
         ok = gen_server:call(?MODULE, {rep_error, RepId, Error}, infinity)
     end.
 
+% Add random delay proportional to the number of replications
+% on current node, in order to prevent a stampede when a source
+% with multiple replication targets fails
+add_error_jitter(RepId, DbName, DocId) ->
+    case ets:info(?REP_TO_STATE, size) of
```

Comment:
Agreed, it means table is deleted or not created yet. We'd want to crash in either case.